src/utils/ExtractTestResults.ts

```diff
@@ -34,23 +37,23 @@ export const extractMCTestResults = (record: DynamoDBRecord): MCRequest[] => {
     )
     .filter(
       (testType) =>
-        testType.testResult === 'pass' || testType.testResult === 'prs',
+        testType.testResult === TestResults.PASS || testType.testResult === TestResults.PRS,
     )
     .filter(
       () =>
         testResultUnmarshall.vehicleType === 'hgv' ||
         testResultUnmarshall.vehicleType === 'psv' ||
         testResultUnmarshall.vehicleType === 'trl',
     )
-    .filter(() => testResultUnmarshall.testStatus === 'submitted')
-    .map((testResult: TestResult) => ({
+    .filter(() => testResultUnmarshall.testStatus === TestStatus.SUBMITTED)
+    .map((testType : TestTypeSchema ) : MCRequest => ({
       vehicleIdentifier:
         testResultUnmarshall.vehicleType === 'trl'
           ? testResultUnmarshall.trailerId
           : testResultUnmarshall.vrm,
-      testDate: isoDateFormatter(testResult.testTypeEndTimestamp),
+      testDate: isoDateFormatter(testType.testTypeEndTimestamp),
       vin: testResultUnmarshall.vin,
-      testResult: calculateTestResult(testResult),
+      testResult: calculateTestResult(testType.testResult as TestResults),
```

Comment:
Why do you need to cast this as TestResults, what other type could this be?

Reply:
```
| {
    [k: string]: unknown;
  }
| TestResults;
```